### PR TITLE
arp: Remove the unnecessary inclusion of nuttx/net/arp

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

and change the remain inclusion to netinet/arp.h
depends on https://github.com/apache/nuttx/pull/7889

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

File: nshlib/nsh_telnetlogin.c [hunk 1]
```diff
@@ -161,9 +162,19 @@ static void nsh_telnettoken(FAR struct console_stdio_s *pstate,
 int nsh_telnetlogin(FAR struct console_stdio_s *pstate)
 {
   char username[16];
-  char password[16];
+  char password[128];
+#ifdef CONFIG_NSH_PLATFORM_CHALLENGE
```
Comment (pkarashchenko) [inline nshlib/nsh_telnetlogin.c:166]: can second commit be moved to a separate PR?
It is a bit out of context of this PR

Comment (pkarashchenko) [reply nshlib/nsh_telnetlogin.c:166]: I see... It is already done

**Comment by pkarashchenko** [comment]
please rebase

File: nshlib/nsh_telnetlogin.c (hunk 1 above)
Comment (xiaoxiang781216) [reply nshlib/nsh_telnetlogin.c]: Sorry, I append the patch to the wrong PR. that patch should belong https://github.com/apache/nuttx-apps/pull/1459.

**Comment by xiaoxiang781216** [comment]
Done.
